### ISSUE TITLE
Manage DKIM keys

### ISSUE THREAD

**Issue by tonioo** [modoboa/modoboa-amavis#8]
_From @tonioo on December 4, 2013 15:45_

**Originally created by Antoine Nguyen on 2012-10-21T10:19:54Z**

Add the possibility to enable DKIM signing from Modoboa. The big issue is: we need to generate amavis config. files :p

_Copied from original issue: tonioo/modoboa#336_

**Comment by tonioo** [comment]
**Posted by Christophe CHAUVET on 2013-07-08T09:39:00Z**

I found OpenDKIM http://www.opendkim.org/ 

More simple to configure than amavis dkim

Regards,

**Comment by tonioo** [comment]
_From @almereyda on June 6, 2014 14:7_

Maybe you want to have a look at how https://github.com/JoshData/mailinabox implements DKIM. Their trick is to run their own DNS for publishing the keys.

**Comment by web-refinery** [comment]
I think adding DNS to the stack should be optionally. Personally I'd rather write an ansible playbook to create/regenerate DKIM's and have that add it to my DNS.

**Comment by RaymondSchnyder** [comment]
is this still a thing? im struggling with DKIM and modoboa at the moment.... :/

**Comment by tonioo** [comment]
@RaymondSchnyder what's your problem ?

**Comment by RaymondSchnyder** [comment]
@tonioo My problem is that every mail i send is marked as spam, i added DMARC and DKIM but it doesnt work whatever i do :/

**Comment by tonioo** [comment]
@RaymondSchnyder Add SPF too and check that your server IP is not blacklisted.

**Comment by RaymondSchnyder** [comment]
My server is not blacklisted and i added SPF from the beginning.

**Comment by jyrkij** [comment]
Quick question: while waiting for official support is it safe to configure DKIM manually with modoboa?

**Comment by tonioo** [comment]
What do you mean? Generate DKIM keys and configure amavis?

**Comment by jyrkij** [comment]
Yes. At least mailinabox overrides configuration files each time it is updated. How does modoboa handle config files?

**Comment by tonioo** [comment]
Actually Modoboa itself does not modify any config file. Configuration files are only touched when you run the installer and it does not support updates yet.

**Comment by jaketame** [comment]
Another one too look at is https://github.com/mailcow/mailcow

https://github.com/mailcow/mailcow/blob/master/opendkim/conf/opendkim.conf
https://github.com/mailcow/mailcow/blob/master/misc/mailcow-dkim-tool

**Comment by manioul** [comment]
This helps a lot! Thx @jichu4n .

**Comment by alisonw** [comment]
Many thanks jichu4n for that routine, which I've just successfully used to set DKIM/DMARC for my domains. One thing though - there doesn't seem to be an obvious route to using Modoboa to generate keys for _alias_ domains. Can they be done manually somehow?

**Comment by tonioo** [comment]
@alisonw Not yet. See https://github.com/modoboa/modoboa/issues/1882.

**Comment by alisonw** [comment]
> Not yet
Ta. I've taken the 'easy' way out by re-creating the aliased domains as standard ones and recreating the users. :-p
